src/export/ExportPCM.cpp

```diff
@@ -409,29 +408,44 @@ class ExportPCM final : public ExportPlugin
 
 };
 
-ExportPCM::ExportPCM()
-   : ExportPlugin()
+ExportPCM::ExportPCM() = default;
+
+int ExportPCM::GetFormatCount() const
 {
-   int selformat; // the index of the format we are setting up at the moment
+   return WXSIZEOF(kFormats) + 1;// + FMT_OTHER
+}
 
-   // Add the "special" formats first
-   for (size_t i = 0; i < WXSIZEOF(kFormats); ++i)
+FormatInfo ExportPCM::GetFormatInfo(int index) const
+{
+   if(index == FMT_OTHER)
    {
-      selformat = AddFormat() - 1;
-      AddExtension(sf_header_extension(kFormats[i].format), selformat);
-      SetFormat(kFormats[i].name, selformat);
-      SetDescription(kFormats[i].desc, selformat);
-      SetCanMetaData(true, selformat);
-      SetMaxChannels(255, selformat);
-   }
+      SF_INFO si = {};
+
+      si.format = LoadOtherFormat() & SF_FORMAT_TYPEMASK;
+      si.format |= LoadEncoding(si.format);
+
+      for (si.channels = 1; sf_format_check(&si); si.channels++)
+      {
+         // just counting
+      }
```

Comment:
Did you neglect `--si.channels;` after the loop?

Reply:
Thanks!